src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -319,45 +341,63 @@ def relative_to_exec_root(path):
     zinc_args.extend(['-compiled-bridge-jar', relative_to_exec_root(compiler_bridge_classpath_entry.path)])
     zinc_args.extend(['-scala-path', ':'.join(scala_path)])
 
-    zinc_args.extend(self._javac_plugin_args(javac_plugin_map))
-    # Search for scalac plugins on the classpath.
-    # Note that:
-    # - We also search in the extra scalac plugin dependencies, if specified.
-    # - In scala 2.11 and up, the plugin's classpath element can be a dir, but for 2.10 it must be
-    #   a jar.  So in-repo plugins will only work with 2.10 if --use-classpath-jars is true.
-    # - We exclude our own classes_dir/jar_file, because if we're a plugin ourselves, then our
-    #   classes_dir doesn't have scalac-plugin.xml yet, and we don't want that fact to get
-    #   memoized (which in practice will only happen if this plugin uses some other plugin, thus
-    #   triggering the plugin search mechanism, which does the memoizing).
-    scalac_plugin_search_classpath = (
-      (set(absolute_classpath) | set(self.scalac_plugin_classpath_elements())) -
-      {ctx.classes_dir.path, ctx.jar_file.path}
-    )
-    zinc_args.extend(self._scalac_plugin_args(scalac_plugin_map, scalac_plugin_search_classpath))
-    if upstream_analysis:
-      zinc_args.extend(['-analysis-map',
-                        ','.join('{}:{}'.format(
-                          relative_to_exec_root(k),
-                          relative_to_exec_root(v)
-                        ) for k, v in upstream_analysis.items())])
-
-    zinc_args.extend(args)
-    zinc_args.extend(self._get_zinc_arguments(settings))
-    zinc_args.append('-transactional')
-
-    compiler_option_sets_args = self.get_merged_args_for_compiler_option_sets(compiler_option_sets)
-
-    # Needed to make scoverage CodeGrid highlighting work
-    if 'scoverage' in scalac_plugin_map.keys():
-      compiler_option_sets_args += ['-S-Yrangepos']
-
-    zinc_args.extend(compiler_option_sets_args)
-
-    if not self._clear_invalid_analysis:
-      zinc_args.append('-no-clear-invalid-analysis')
-
-    if not zinc_file_manager:
-      zinc_args.append('-no-zinc-file-manager')
+    from pants.backend.jvm.tasks.jvm_compile.rsc.rsc_compile import RscCompileContext
+    use_youtline = ctx is RscCompileContext
+    if use_youtline:
+      wartremover_args = [
+        f"-S-Xplugin:{self._wartremover_classpath[0]}",
```

Comment:
I know that this is existing code, but given that this is injected into the compilation phase (where we [already have support for plugins](https://www.pantsbuild.org/scalac_plugins.html)), it feels like it would be easier to have users configure it if they want it?

Reply:
removed the plugin dependency